src/lib/components/resources/Gms2Sound.ts

```diff
@@ -53,16 +53,42 @@ export class Gms2Sound extends Gms2ResourceBase {
     this.save();
   }
 
+  get channels(){
+    return this.yyData.type;
+  }
+
   set channels(channel: SoundChannel){
+    this.setChannel(channel);
+  }
+
+  private setChannel(channel: SoundChannel){
     this.yyData.type = channel;
     this.save();
   }
 
+  setChannelByString(channelAsString: keyof typeof SoundChannel){
+    const channel = SoundChannel[channelAsString];
+    this.setChannel(channel);
+  }
+
+  get compression(){
+    return this.yyData.compression;
+  }
+
   set compression(level: SoundCompression){
+    this.setCompreesion(level);
+  }
+
+  private setCompreesion(level: SoundCompression){
     this.yyData.compression = level;
     this.save();
   }
 
+  setCompressionByString(levelAsString: keyof typeof SoundCompression){
```

Comment:
Since you've made the other function private, this one should have a name that is more intuitive. Calling this one `setCompression` , and the other `setCompressionByEnum` might make more sense.

However, I'd clean all of this up to be simpler: Have the setter allow *either* strings or numbers, have the getter return the string, and then add another getter or function like `channelsAsIndex` that returns the numeric form. This would make for a very easy to use API.